sql/logictest: TestLogic failed

sql/logictest.TestLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3724574&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3724574&tab=artifacts#/) on master @ [728726cdce49f53b13e3d87c209fae628d441290](https://github.com/cockroachdb/cockroach/commits/728726cdce49f53b13e3d87c209fae628d441290):


```
=== RUN   TestLogic/local-spec-planning
    --- FAIL: TestLogic/local-spec-planning (0.00s)
=== RUN   TestLogic/local-spec-planning/alter_column_type/alter_column_type_general
=== CONT  TestLogic/local-spec-planning/alter_column_type/alter_column_type_general
    logic.go:2570: 
        
        testdata/logic_test/alter_column_type:343: SELECT x, y FROM t15 ORDER BY x
        expected success, but found
        (XX000) internal error: runtime error: index out of range [-1]
        catch.go:29: in ShouldCatch()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/util/errorutil/catch.go:29: ShouldCatch()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:205: func1()
        runtime/panic.go:965: gopanic()
        runtime/panic.go:88: goPanicIndex()
        github.com/cockroachdb/cockroach/pkg/sql/opt_catalog.go:1039: Column()
        github.com/cockroachdb/cockroach/pkg/sql/opt_catalog.go:1359: Column()
        github.com/cockroachdb/cockroach/pkg/sql/opt/table_meta.go:214: IndexColumns()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/scan_index_iter.go:279: ForEachStartingAfter()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/scan_index_iter.go:209: ForEach()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/scan_funcs.go:45: GenerateIndexScans()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:84: exploreScan()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:18: exploreGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.go:182: exploreGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:468: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:620: optimizeEnforcer()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:586: enforceProps()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:495: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:455: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:225: Optimize()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:525: buildExecMemo()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:194: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1264: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:962: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:700: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:129: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1748: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1750: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1672: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:683: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:666: func1()
        runtime/asm_amd64.s:1371: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2319: 
         pq: internal error: runtime error: index out of range [-1]
    logic.go:1863: 
         testdata/logic_test/alter_column_type:349: too many errors encountered, skipping the rest of the input
--- done: testdata/logic_test/alter_column_type with config local-spec-planning: 72 tests, 3 failures
            --- FAIL: TestLogic/local-spec-planning/alter_column_type/alter_column_type_general (3.12s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #63262 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-20.2]
- #61657 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-21.1]
- #58269 sql/logictest: TestLogic failed [C-test-failure O-robot T-sql-schema branch-release-20.1]
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/logictest.TestLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3724930&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3724930&tab=artifacts#/) on master @ [f82ff534856738d5385073167d048feafb0b4f3e](https://github.com/cockroachdb/cockroach/commits/f82ff534856738d5385073167d048feafb0b4f3e):


```
        testdata/logic_test/alter_column_type:343: SELECT x, y FROM t15 ORDER BY x
        expected success, but found
        (XX000) internal error: runtime error: index out of range [-1]
        catch.go:29: in ShouldCatch()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/util/errorutil/catch.go:29: ShouldCatch()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:205: func1()
        runtime/panic.go:965: gopanic()
        runtime/panic.go:88: goPanicIndex()
        github.com/cockroachdb/cockroach/pkg/sql/opt_catalog.go:1039: Column()
        github.com/cockroachdb/cockroach/pkg/sql/opt_catalog.go:1359: Column()
        github.com/cockroachdb/cockroach/pkg/sql/opt/table_meta.go:214: IndexColumns()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/scan_index_iter.go:279: ForEachStartingAfter()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/scan_index_iter.go:209: ForEach()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/scan_funcs.go:45: GenerateIndexScans()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:84: exploreScan()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:18: exploreGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.go:182: exploreGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:468: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:620: optimizeEnforcer()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:586: enforceProps()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:495: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:455: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:225: Optimize()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:525: buildExecMemo()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:194: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1264: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:962: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:700: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:129: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1748: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1750: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1672: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:683: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:666: func1()
        runtime/asm_amd64.s:1371: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2319: 
         pq: internal error: runtime error: index out of range [-1]
    logic.go:1863: 
         testdata/logic_test/alter_column_type:349: too many errors encountered, skipping the rest of the input
--- done: testdata/logic_test/alter_column_type with config fakedist: 72 tests, 3 failures
            --- FAIL: TestLogic/fakedist/alter_column_type/alter_column_type_general (2.48s)
=== RUN   TestLogic/fakedist/alter_column_type
=== PAUSE TestLogic/fakedist/alter_column_type
=== CONT  TestLogic/fakedist/alter_column_type
        --- FAIL: TestLogic/fakedist/alter_column_type (6.66s)
=== RUN   TestLogic/fakedist
    --- FAIL: TestLogic/fakedist (0.00s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #63262 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-20.2]
- #61657 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-21.1]
- #58269 sql/logictest: TestLogic failed [C-test-failure O-robot T-sql-schema branch-release-20.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/logictest.TestLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3734018&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3734018&tab=artifacts#/) on master @ [eeec6ff9f0053cfc105fb735b19fe0608683c996](https://github.com/cockroachdb/cockroach/commits/eeec6ff9f0053cfc105fb735b19fe0608683c996):


```
        expected success, but found
        (XX000) internal error: runtime error: index out of range [-1]
        catch.go:29: in ShouldCatch()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/util/errorutil/catch.go:29: ShouldCatch()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:205: func1()
        runtime/panic.go:965: gopanic()
        runtime/panic.go:88: goPanicIndex()
        github.com/cockroachdb/cockroach/pkg/sql/opt_catalog.go:1039: Column()
        github.com/cockroachdb/cockroach/pkg/sql/opt_catalog.go:1359: Column()
        github.com/cockroachdb/cockroach/pkg/sql/opt/table_meta.go:214: IndexColumns()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/scan_index_iter.go:279: ForEachStartingAfter()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/scan_index_iter.go:209: ForEach()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/scan_funcs.go:45: GenerateIndexScans()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:84: exploreScan()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:18: exploreGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.go:182: exploreGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:468: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:620: optimizeEnforcer()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:586: enforceProps()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:495: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:455: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:225: Optimize()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:525: buildExecMemo()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:194: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1264: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:962: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:700: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:129: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1748: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1750: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1672: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:683: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:666: func1()
        runtime/asm_amd64.s:1371: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2400: 
         pq: internal error: runtime error: index out of range [-1]
    logic.go:1946: 
         testdata/logic_test/alter_column_type:341: too many errors encountered, skipping the rest of the input
--- done: testdata/logic_test/alter_column_type with config local-spec-planning: 71 tests, 3 failures
--- progress: testdata/logic_test/fk: 718 statements/queries
            --- FAIL: TestLogic/local-spec-planning/alter_column_type/alter_column_type_general (3.54s)
=== RUN   TestLogic/local-spec-planning/alter_column_type
=== PAUSE TestLogic/local-spec-planning/alter_column_type
=== CONT  TestLogic/local-spec-planning/alter_column_type
        --- FAIL: TestLogic/local-spec-planning/alter_column_type (9.16s)
=== RUN   TestLogic/local-spec-planning
    --- FAIL: TestLogic/local-spec-planning (0.01s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #63262 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-20.2]
- #61657 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-21.1]
- #58269 sql/logictest: TestLogic failed [C-test-failure O-robot T-sql-schema branch-release-20.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/logictest.TestLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3734511&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3734511&tab=artifacts#/) on master @ [eeec6ff9f0053cfc105fb735b19fe0608683c996](https://github.com/cockroachdb/cockroach/commits/eeec6ff9f0053cfc105fb735b19fe0608683c996):


```
        expected success, but found
        (XX000) internal error: runtime error: index out of range [-1]
        catch.go:29: in ShouldCatch()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/util/errorutil/catch.go:29: ShouldCatch()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:205: func1()
        runtime/panic.go:965: gopanic()
        runtime/panic.go:88: goPanicIndex()
        github.com/cockroachdb/cockroach/pkg/sql/opt_catalog.go:1039: Column()
        github.com/cockroachdb/cockroach/pkg/sql/opt_catalog.go:1359: Column()
        github.com/cockroachdb/cockroach/pkg/sql/opt/table_meta.go:214: IndexColumns()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/scan_index_iter.go:279: ForEachStartingAfter()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/scan_index_iter.go:209: ForEach()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/scan_funcs.go:45: GenerateIndexScans()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:84: exploreScan()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:18: exploreGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.go:182: exploreGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:468: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:620: optimizeEnforcer()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:586: enforceProps()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:495: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:455: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:225: Optimize()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:525: buildExecMemo()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:194: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1264: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:962: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:700: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:129: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1748: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1750: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1672: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:683: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:666: func1()
        runtime/asm_amd64.s:1371: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2400: 
         pq: internal error: runtime error: index out of range [-1]
    logic.go:1946: 
         testdata/logic_test/alter_column_type:341: too many errors encountered, skipping the rest of the input
--- done: testdata/logic_test/alter_column_type with config local-spec-planning: 71 tests, 3 failures
--- progress: testdata/logic_test/upsert: 258 statements/queries
            --- FAIL: TestLogic/local-spec-planning/alter_column_type/alter_column_type_general (2.94s)
=== RUN   TestLogic/local-spec-planning/alter_column_type
=== PAUSE TestLogic/local-spec-planning/alter_column_type
=== CONT  TestLogic/local-spec-planning/alter_column_type
        --- FAIL: TestLogic/local-spec-planning/alter_column_type (6.39s)
=== RUN   TestLogic/local-spec-planning
    --- FAIL: TestLogic/local-spec-planning (0.01s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #63262 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-20.2]
- #61657 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-21.1]
- #58269 sql/logictest: TestLogic failed [C-test-failure O-robot T-sql-schema branch-release-20.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/alter_column_type.go
+++ b/pkg/sql/alter_column_type.go
@@ -232,6 +232,13 @@ func alterColumnTypeGeneral(
 				return colInIndexNotSupportedErr
 			}
 		}
+		if !idx.Primary() {
+			for i := 0; i < idx.NumSecondaryStoredColumns(); i++ {
+				if idx.GetStoredColumnID(i) == col.GetID() {
+					return colInIndexNotSupportedErr
+				}
+			}
+		}
 	}
 
 	// Disallow ALTER COLUMN TYPE general inside an explicit transaction.

--- a/pkg/sql/alter_column_type_test.go
+++ b/pkg/sql/alter_column_type_test.go
@@ -332,7 +332,7 @@ func TestSchemaChangeBeforeAlterColumnType(t *testing.T) {
 			},
 		},
 	}
-
+	defer close(waitBeforeContinuing)
 	s, db, _ := serverutils.StartServer(t, params)
 	sqlDB := sqlutils.MakeSQLRunner(db)
 	defer s.Stopper().Stop(ctx)
@@ -353,7 +353,8 @@ ALTER TABLE t.test ALTER PRIMARY KEY USING COLUMNS (x);
 
 	<-swapNotification
 
-	expected := "pq: unimplemented: table test is currently undergoing a schema change"
+	expected := "pq: unimplemented: ALTER COLUMN TYPE requiring rewrite of on-disk data is currently not " +
+		"supported for columns that are part of an index"
 	sqlDB.ExpectErr(t, expected, `
 SET enable_experimental_alter_column_type_general = true;
 ALTER TABLE t.test ALTER COLUMN y TYPE STRING;`)

--- a/pkg/sql/logictest/testdata/logic_test/alter_column_type
+++ b/pkg/sql/logictest/testdata/logic_test/alter_column_type
@@ -330,20 +330,14 @@ CREATE TABLE uniq (x INT, y INT, UNIQUE WITHOUT INDEX (x, y))
 statement error pq: unimplemented: ALTER COLUMN TYPE for a column that has a constraint is currently not supported
 ALTER TABLE uniq ALTER COLUMN x TYPE STRING
 
-# Ensure we can change the column type of a column stored in a secondary index.
+# Ensure we cannot change the column type of a column stored in a secondary index.
 statement ok
 CREATE TABLE t15 (x INT, y INT);
 CREATE INDEX ON t15 (x) STORING (y);
 INSERT INTO t15 VALUES (1, 1), (2, 2)
 
-statement ok
+statement error pq: unimplemented: ALTER COLUMN TYPE requiring rewrite of on-disk data is currently not supported for columns that are part of an index
 ALTER TABLE t15 ALTER COLUMN y TYPE STRING;
-
-query IT
-SELECT x, y FROM t15 ORDER BY x
-----
-1  1
-2  2
 
 # Ensure ALTER COLUMN TYPE works for collated strings.
 statement ok

